x-pack/plugins/apm/public/components/app/transaction_details/waterfall_with_summary/waterfall_container/Waterfall/waterfall_item.tsx

```diff
@@ -182,6 +187,9 @@ export function WaterfallItem({
     }
   );
 
+  const isCompositeSpan = item.docType === 'span' && (item as IWaterfallSpan).doc.span.composite !== undefined;
```

Comment:
Do you really need to manually convert item to `IWaterfallSpan`. I think Typescript should do it automatically because of `item.docType === 'span'`

```suggestion
  const isCompositeSpan = item.docType === 'span' && item.doc.span.composite;
```